vip-datamanagement/src/main/java/fr/insalyon/creatis/vip/datamanager/server/business/LFCPermissionBusiness.java

```diff
@@ -69,33 +69,36 @@ public LFCPermissionBusiness(DataManagerBusiness dataManagerBusiness, LfcPathsBu
         READ, UPLOAD, DELETE
     }
 
-    public void checkPermission(User user, String path, LFCAccessType LFCAccessType)
+    public Boolean isLFCPathAllowed(User user, String path, LFCAccessType LFCAccessType, Boolean disableAdminAccess)
             throws BusinessException {
-        // TODO : verify there is no problem with ".." (normalize if its the case)
-        checkRootPermission(path, LFCAccessType);
-        // Root is always filtered so always permited
-        if (path.equals(ROOT)) return;
+        // normalize to remove "..".
+        path = Paths.get(path).normalize().toString();
+        if ( ! isRootOK(user, path, LFCAccessType)) return false;
+
+        // Root is always filtered so always permitted
+        if (path.equals(ROOT)) return true;
+        // do not delete synchronized stuff
+        if (LFCAccessType == LFCPermissionBusiness.LFCAccessType.DELETE
+            && isSynchronized(user, path)) {
+            return false;
+        }
         // else it all depends of the first directory
         String firstDir = getFirstDirectoryName(path);
         // always can access its home and its trash
-        if (firstDir.equals(USERS_HOME)) return;
-        if (firstDir.equals(TRASH_HOME)) return;
-        // currently no admin access is possible via the api for security reasons
+        if (firstDir.equals(USERS_HOME)) return true;
+        if (firstDir.equals(TRASH_HOME)) return true;
         if (firstDir.equals(USERS_FOLDER) || firstDir.equals(VO_ROOT_FOLDER)) {
-            logger.error("Trying to access admin folders from api : {}", path);
-            throw new BusinessException("Unauthorized LFC access");
+            // restricted to admin
+            return isAdminAllowed(user, path, disableAdminAccess);
         }
         // else it should be a group folder
         if (!firstDir.endsWith(GROUP_APPEND)) {
-            logger.error("Unexpected lfc access to: " + firstDir);
-            throw new BusinessException("Unauthorized LFC access");
+            logger.error("({}) Wrong lfc access to '{}'",
+                    user.getEmail(), path);
+            return false;
         }
         String groupName = firstDir.substring(0,firstDir.length()-GROUP_APPEND.length());
-        checkGroupPermission(user, groupName, LFCAccessType);
-        if (LFCAccessType == LFCPermissionBusiness.LFCAccessType.DELETE) {
-            checkSynchronizedDirectories(user, path);
-        }
-        // all check passed : all good !
+        return isGroupAllowed(user, groupName, LFCAccessType);
```

Comment:
So admin is granted access (unless disableAdminAccess is true) to all folders except Group folders (since isGroupAllowed does not consider admins specifically). 
Personally, I would like to have access to group folders if I have admin rights.

Reply:
Indeed I forgot admin access for groups.